File: legacy/packages/core/src/actions/monitor/health.ts
import { z } from "zod";
import { ActionHandler } from "../../types.js";

export const HealthSchema = z.object({
    action: z.literal("health").describe("Health action - quick database connectivity and version check"),
});

export const healthHandler: ActionHandler<typeof HealthSchema> = {
    schema: HealthSchema,
    handler: async (_params, context) => {
        const sql = "SELECT version(), current_database(), now()";
        const result = await context.executor.execute(sql);
        return {
            status: "healthy",
            database: result.rows[0].current_database,
            version: result.rows[0].version,
            server_time: result.rows[0].now,
        };
    },
};

File: coldquery/actions/monitor/health.py
import json
from typing import Dict, Any
from coldquery.core.context import ActionContext, resolve_executor

async def health_handler(params: Dict[str, Any], context: ActionContext) -> str:
    """Database health check."""
    executor = await resolve_executor(context, None)

    try:
        result = await executor.execute("SELECT 1 AS health_check")
        if result.row_count == 1 and result.rows[0].get("health_check") == 1:
            health_status = {"status": "ok"}
        else:
            health_status = {"status": "error", "reason": "Health check query failed"}
    except Exception as e:
        health_status = {"status": "error", "reason": str(e)}

    return json.dumps(health_status)
